src/problems/ode_problems.jl

```diff
@@ -108,15 +108,18 @@ struct ODEProblem{uType, tType, isinplace, P, F, K, PT} <:
     kwargs::K
     """An internal argument for storing traits about the solving process."""
     problem_type::PT
+    """Whether the output has all saved states."""
+    dense_output::Bool
     @add_kwonly function ODEProblem{iip}(f::AbstractODEFunction{iip},
                                          u0, tspan, p = NullParameters(),
                                          problem_type = StandardODEProblem();
+                                         dense_output = true,
                                          kwargs...) where {iip}
         _tspan = promote_tspan(tspan)
         new{typeof(u0), typeof(_tspan),
             isinplace(f), typeof(p), typeof(f),
             typeof(kwargs),
-            typeof(problem_type)}(f, u0, _tspan, p, kwargs, problem_type)
+            typeof(problem_type)}(f, u0, _tspan, p, kwargs, problem_type, dense_output)
```

Comment:
Why do we need this?

Reply:
otherwise it breaks observed for odae and others

Reply:
Why would it break that?

Reply:
It was too hard to catch all the cases and not break everything without doing this. I would need some kind of signal from problem construction time to post solve regrardless, to signal what to call the observed functions with

Reply:
What signal do you need other than save_idxs?

Reply:
we don't have save_idxs as problem construction time

Reply:
Yeah, but maybe we should.

Reply:
@ChrisRackauckas thoughts on the above?